pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -391,6 +399,134 @@ void writeMetadata()
     properties.save();
   }
 
+  /**
+   * Check for the validity of segment start and end time
+   * @param startTime segment start time
+   * @param endTime segment end time
+   * @param segmentName segment name
+   */
+  private void checkTime(final SegmentGeneratorConfig config, final Object startTime, final Object endTime,
+      final String segmentName) {
+    if (!config.isCheckTimeColumnValidityDuringGeneration()) {
+      return;
+    }
+
+    if (startTime == null || endTime == null) {
+      throw new RuntimeException("Expecting non-null start/end time for segment: " + segmentName);
+    }
+
+    long start;
+    long end;
+
+    if (startTime instanceof Long && endTime instanceof Long) {
```

Comment:
I'm thinking whether we can firstly check `startTime` ad `endTime` are of the same class, then use switch block to differentiate the cases on either one of the objects. By doing so we don't need to keep a list of `if else block` here. But you can decide.

Reply:
Done